### PR TITLE
Changed the build tool from Webpack to Vite

### PR DESCRIPTION
**Pull request by LuisdelaVega**
Changed the build tool from Webpack to Vite.

## Description

This migration is an attempt to modernize the Comet starter app by using a build tool like Vite. This will allow us to keep up with modern web development standards while taking advantage of the immediate impact to the development experience and speed that Vite provides.

The React team [mainly suggest](https://react.dev/learn/start-a-new-react-project) using what they call "Production-grade React frameworks" when creating a new project, but they have recommended the use of Vite, most noticeably on their page about [adding React to an existing project](https://react.dev/learn/add-react-to-an-existing-project).

## Related Issue

N/A

## Motivation and Context

N/A

## How Has This Been Tested?

Locally. Running `npm run dev`, `npm run test`, and `npm run e2e`.

## Screenshots (if appropriate):

N/A

### CODE REVIEW

File: .gitattributes
Comment (jbouder) [inline .gitattributes:2]: 👍

File: .vscode/extensions.json
Comment (jbouder) [reply .vscode/extensions.json:14]: Also, going to suggest we add this one: `firsttris.vscode-jest-runner`

File: src/styles.scss
Comment (jbouder) [inline src/styles.scss:1]: Can this be removed?

File: src/types.d.ts
Comment (jbouder) [inline src/types.d.ts:3]: Assuming these are no longer needed? If so, lets remove

File: vite.config.ts
Comment (jbouder) [inline vite.config.ts:19]: I tend to like the app opening when the dev environment starts up. Can you add `open: true,` here?

**Review by jbouder — CHANGES_REQUESTED** [review]
In general, really great work! Was able to verify the app loads super fast and can still do a production build. Also ran all quality checks successfully. Please review the comments provided below.

Also, once all comments are addressed, we will want to get a review from Scott Blair for any feedback.

File: .prettierrc
Comment (jbouder) [reply .prettierrc:3]: One other major thing, I prefer single quotes. If you want to go with double, will need a justification. And if we go with double, we'll want to implement on the Comet repo as well for consistency.

File: .vscode/settings.json
Comment (ldelavegalopez-metrostar) [reply .vscode/settings.json:5]: I use it too. It's a part of my user settings. I commented it out and made a comment on how to enable it in the README because feel like this one is a bit of personal preference. Some people might not like their editor to format their code like this. Dunno, I'm happy to uncomment it if you think we should.

File: README.md
Comment (ldelavegalopez-metrostar) [reply README.md:10]: Oh yeah. And the Reack Query one.

File: src/utils/axios.ts
Comment (ldelavegalopez-metrostar) [reply src/utils/axios.ts:4]: I'm using `EnvironmentPlugin` to expose these values through `process.env`. This is useful because Jest doesn't have access to these values with the way Vite does it by default.

File: .vscode/settings.json
Comment (jbouder) [reply .vscode/settings.json:5]: Works for me

File: src/utils/axios.ts
Comment (jbouder) [reply src/utils/axios.ts:4]: Sounds good.

File: .vscode/extensions.json
Comment (ldelavegalopez-metrostar) [reply .vscode/extensions.json:14]: Have you tried `orta.vscode-jest`? It's the one I added to the recommendations. I used `firsttris.vscode-jest-runner` for many years until I switched and to the Orta one and I haven't looked back. But I'm not going to die on this hill lol. Whatever you prefer I'll add to the recommendations.

File: .prettierrc
Comment (ldelavegalopez-metrostar) [reply .prettierrc:3]: 👍 I'll change it to single quotes.

I have absolutely 0 opinion on the single vs double quote discussion. I just use whatever Prettier defaults to.

File: .gitignore
Comment (LuisdelaVega) [reply .gitignore]: Can you review the changes to this file and let me know if there's anything else that stands out?

**Comment by LuisdelaVega** [comment]
Just making a note here that I will squash all of these commits before merging.

**Review by jbouder — APPROVED** [review]
Great work! Thank you for running this one down.

**Comment by jbouder** [comment]
@sblair-metrostar can you review this change from WebPack to Vite to see if anything jumps out as potential pipeline related concerns?

File: .gitignore
Comment (jbouder) [reply .gitignore]: Looks good

**Comment by sblair-metrostar** [comment]
> @sblair-metrostar can you review this change from WebPack to Vite to see if anything jumps out as potential pipeline related concerns?

Unless it creates a dependency on something other than Node 18 and npm/yarn to run the build, I wouldn't expect it to matter much from a pipeline perspective.

**Pull request merged**
